GREEDY/DataManagers/DataConverter.cs

```diff
@@ -1,122 +1,127 @@
                         });
+                        sublist = receipt.LinesOfText.GetRange(i + 1, receipt.LinesOfText.Count - i - 1);
+                        break;
                     }
-                    return itemList;
                 }
-                return itemList;
+                else
+                {
+                    previous = receipt.LinesOfText[i];
+                }
             }
-            else
+
+            //working with text
+            previous = String.Join(Environment.NewLine, sublist);
+            previous = Regex.Replace(previous, @"\r", "");
+            previous = Regex.Replace(previous, @"\n", " ");
+            previous = Regex.Replace(previous, "›", ",");
+            previous = Regex.Replace(previous, @"(\d+[\.\,]\d{2}.[A|E|B|F|N|C]{1}(\b|\.))", "$1" + Environment.NewLine);
+
+            MatchCollection match = Regex.Matches(previous, pattern, RegexOptions.Multiline);
+            if (match.Count != 0)
             {
-                throw new NotImplementedException();
+                foreach (Match m in match)
+                {
+                    itemList.Add(new Item
+                    {
+                        Name = m.Groups[1].Value.Replace("\n", string.Empty),
+                        Price = decimal.Parse(m.Groups[2].Value.Replace(".", ",")),
+                        Category = ItemCategorization.CategorizeSingleItem(m.Groups[1].Value)
+                    });
+                }
             }
-        }
-    
-        // TODO
-        // this doesnt belong in this class, can be moved to a static method, maybe an extension method
-        public XElement ListToXml(List<Item> items)
-        {
-            XElement xmlElements = new XElement("items",
-                items.Select(i => new XElement("item",
-                new XAttribute("Name", i.Name),
-                new XAttribute("Price", i.Price),
-                new XAttribute("Category", i.Category))));
-            return xmlElements;
-        }
 
-        // TODO
-        // this doesnt belong in this class, can be moved to a static method, maybe an extension method
-        public DataTable ListToDataTable(List<Item> items)
-        {
-            DataTable dataTable = new DataTable();
-            dataTable.Columns.Add("Name");
-            dataTable.Columns.Add("Price");
-            dataTable.Columns.Add("Category");
+            Console.WriteLine(receipt.Shop.Name);
+            Console.WriteLine(receipt.Date);
 
-            foreach (Item item in items)
+            //if no item was created. Need to create a massage for user
+            if (itemList.Count == 0)
             {
-                DataRow dr = dataTable.NewRow();
-                dr[0] = item.Name;
-                dr[1] = item.Price;
-                dr[2] = item.Category;
-                dataTable.Rows.Add(dr);
+                Console.WriteLine("Nepavyko nuskaityti cekio");
```

Comment:
This shouldn't be a console write line. If no item was created there are two options one is to return null other is throw an exception. Then the controller that is calling this method should deal with it (I believe returning null would be better because exceptions are for errors and it is not really an error that no items have been read)

Reply:
We has talked about it. We need to create information dialog for a user. This is just for testing. Can you create an informative error windiw for user? :)